src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -34,7 +34,7 @@ public static string GetFullPath(string path)
 
             // We would ideally use realpath to do this, but it resolves symlinks, requires that the file actually exist,
             // and turns it into a full path, which we only want if fullCheck is true.
-            string collapsedString = PathInternal.RemoveRelativeSegments(path, PathInternal.GetRootLength(path));
+            string collapsedString = Path.RemoveRedundantSegments(path.AsSpan());
```

Comment:
Having numbers performance numbers before/after this change for the existing Path APIs would be useful. https://github.com/dotnet/performance has micro-benchmarks for Path APIs.

On Windows, this change is replacing ~100 line `RemoveRelativeSegments` method with ~400 lines in a new `RedundantSegmentHelper` type and multiple methods. That's a bit concerning for an API that was advertised as already implemented as an internal API (https://github.com/dotnet/runtime/issues/2162#issue-554929427).

Reply:
No problem. I'm working on getting the performance numbers.

Some things to keep in mind:
- The reason why I had to rewrite everything is because the original internal method was unable to handle all the Windows edge cases, particularly with device prefixes.
- I put everything in a new type because it would help having the code organized. Should I put everything in `Path.cs` to prevent creating a new type, even if that class gets cluttered?
- I split the functionality into all those methods to make the code more readable. Should I instead put as much code as possible into fewer methods, even if we sacrifice readability?

Reply:
@jkotas / @carlossanlop I recommend we:

1. Capture the numbers to see how much of a regression we introduce
2. Decide if the regression is either A) acceptable, B) within reason, but should be tuned, or C) not within reason
3. If it's within reason but should be tuned, file a perf regression issue to fix after Preview 8 and before RC2 (with other perf regression issues)
4. If it the regression ends up being so egregious that we aren't confident we'd be able to tune it back within reason, reevaluate this PR